src/tools/miri/tests/fail/never_transmute_void.rs

```diff
@@ -10,11 +10,13 @@ mod m {
     pub struct Void(VoidI);
 
     pub fn f(v: Void) -> ! {
-        match v.0 {} //~ ERROR: entering unreachable code
+        match v.0 {}
     }
 }
 
 fn main() {
     let v = unsafe { std::mem::transmute::<(), m::Void>(()) };
-    m::f(v); //~ NOTE: inside `main`
+    //~^ ERROR: transmuting to uninhabited type
```

Comment:
Note that MIRI is now detecting this earlier than it did before.  That seems good to me, but I figured I'd mention it in case that's somehow wrong.

(The [`never_say_never` test](https://github.com/rust-lang/rust/blob/c0f37075b6d5e887c0d62853cfbd60bb0010b58d/src/tools/miri/tests/fail/never_say_never.rs), which doesn't go through `transmute`, will continue to test the "it looks like unreachable" form of this UB from using uninhabited types.)

Reply:
This test explicitly passes `-Zmiri-disable-validation`, so detecting it earlier is odd. With default flags Miri already detects this earlier today.